src/lib/selector/fast-auth-wallet.ts

```diff
@@ -0,0 +1,310 @@
+          nearAPI.utils.serialize.base_decode(block.header.hash),
+        );
+      }),
+    );
+  };
+
+  return {
+    async signIn({ contractId, methodNames, successUrl, failureUrl, email, accountId, isRecovery }: any) {
+      const existingAccounts = await getAccounts();
+
+      if (existingAccounts.length) {
+        return existingAccounts;
+      }
+
+      await _state.wallet.requestSignIn({
+        contractId,
+        methodNames,
+        successUrl,
+        failureUrl,
+        email,
+        accountId,
+        isRecovery,
+      });
+
+      return getAccounts();
+    },
+
+    async signOut() {
+      if (_state.wallet.isSignedIn()) {
+        _state.wallet.signOut();
+      }
+    },
+
+    async getAccounts() {
+      return getAccounts();
+    },
+
+    async verifyOwner() {
+      throw new Error(`Method not supported by ${metadata.name}`);
+    },
+
+    async signAndSendTransaction({ receiverId, actions, signerId }) {
+      const account = _state.wallet.account();
+
+      const { accessKey } = await account.findAccessKey(receiverId as string, []);
+
+      const needsFAK =
+        accessKey.permission !== 'FullAccess' && accessKey.permission.FunctionCall.receiver_id !== receiverId;
+
+      if (needsFAK) {
+        const { signer, networkId, provider } = account.connection;
+        const block = await provider.block({ finality: 'final' });
+        const localKey = await signer.getPublicKey(account.accountId, networkId);
+        const txAccessKey = await account.accessKeyForTransaction(
+          receiverId as string,
+          actions.map(createAction),
+          localKey,
+        );
+        const transaction = nearAPI.transactions.createTransaction(
+          account.accountId,
+          nearAPI.utils.PublicKey.from(txAccessKey.public_key),
+          receiverId as string,
+          txAccessKey.access_key.nonce + 1,
+          actions.map(createAction),
+          nearAPI.utils.serialize.base_decode(block.header.hash),
+        );
+        const arg = {
+          transactions: [transaction],
+        };
+        const closeDialog = await _state.wallet.requestSignTransactions(arg);
+        const listener = (e: MessageEvent) => {
+          const shouldRemove = processSignedDelegates(params.relayerUrl, closeDialog, e);
+          if (shouldRemove) {
+            window.removeEventListener('message', listener, false);
+          }
+        };
+        window.addEventListener('message', listener);
+      } else {
+        const signedDelegate = await account.signedDelegate({
+          actions: actions.map((action) => createAction(action)),
+          blockHeightTtl: 60,
+          receiverId: receiverId as string,
+        });
+
+        await fetch(params.relayerUrl, {
+          method: 'POST',
+          mode: 'cors',
+          body: JSON.stringify(Array.from(encodeSignedDelegate(signedDelegate))),
+          headers: new Headers({ 'Content-Type': 'application/json' }),
+        });
```

Comment:
error handling here too

Reply:
Per above these aren't currently addressed in prod and the caller here is the VM, will address in a follow up PR